tests/test_pandas.py

```diff
@@ -130,6 +130,49 @@ def test_csv():
     assert_equal(df.columns.tolist(), df2.columns.tolist())
     assert_equal(df.dtypes.tolist(), df2.dtypes.tolist())
 
+    with NamedTemporaryFile(suffix='.csv', mode='w+') as f:
+        df.to_csv(f, index=False, columns=['name', 'act'])
+        f.seek(0)
+        df2 = opd.read_csv(f)
+    assert_equal(df[['name', 'act']].shape, df2.shape)
+    assert_equal(df[['name', 'act']].columns.tolist(), df2.columns.tolist())
+    assert_equal(df[['name', 'act']].dtypes.tolist(), df2.dtypes.tolist())
+
+
+def test_excel():
+    # just check if it doesn't fail
+    df = opd.read_sdf(os.path.join(test_data_dir, 'data/dude/xiap/actives_docked.sdf'),
+                      columns=['mol', 'name', 'chembl_id', 'dude_smiles', 'act'])
+    df['act'] = df['act'].astype(float)
+    df['name'] = df['name'].astype(int)
+    with NamedTemporaryFile(suffix='.xls', mode='w') as f:
+        df.to_excel(f.name, index=False)
```

Comment:
All this methods should work with `path_or_buf` so `f` should work and tested. If `path_or_buf` is empty, `str` should be returned, thus might be done in one line (altogether 3 tests per 'writer').

Reply:
We can (and should) do such tests for CSV writer, but API for Excel is different and only file name or ExcelWriter object are supported (see [docs](https://pandas.pydata.org/pandas-docs/stable/generated/pandas.DataFrame.to_excel.html)) Btw we currently don't support ExcelWriter, but it's an easy fix